gateway/api/views.py

```diff
@@ -153,10 +102,13 @@ def retrieve(self, request, pk=None):  # pylint: disable=arguments-differ
         job: Job = get_object_or_404(queryset, pk=pk)
         serializer = self.get_serializer(job)
         if job.compute_resource:
-            ray_client = JobSubmissionClient(job.compute_resource.host)
-            ray_job_status = ray_client.get_job_status(job.ray_job_id)
-            job.status = ray_job_status_to_model_job_status(ray_job_status)
-            job.save()
+            try:
+                ray_client = JobSubmissionClient(job.compute_resource.host)
+                ray_job_status = ray_client.get_job_status(job.ray_job_id)
+                job.status = ray_job_status_to_model_job_status(ray_job_status)
```

Comment:
The status update here may cause skipping the capturing logs at line 21 in update_jobs_statuses.py when the status change from pending to succeeded.

Reply:
Good catch!